### PR TITLE
New dev host options

### PR DESCRIPTION
**Pull request by mipearson**
TODO:

* [ ] Manual testing
* [ ] README documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,6 +70,16 @@ Have a look at the files in the `examples` directory. Of note:
   * Webpacked assets will be compiled to `public/webpack`
   * The manifest file is named `manifest.json`
 
+#### Dynamic host
+
+To have the host evaluated at request-time, set `host` to a proc:
+
+```ruby
+config.webpack.dev_server.host = proc { request.host }
```
Comment (joevandyk) [inline README.md]: maybe `proc { |request| request.host }` ?

File: lib/webpack/rails/helper.rb [hunk 2]
```diff
@@ -18,9 +18,11 @@ def webpack_asset_paths(source, extension: nil)
         paths = Webpack::Rails::Manifest.asset_paths(source)
         paths = paths.select {|p| p.ends_with? ".#{extension}" } if extension
 
-        host = ::Rails.configuration.webpack.dev_server.host
         port = ::Rails.configuration.webpack.dev_server.port
 
+        host = ::Rails.configuration.webpack.dev_server.host
+        host = instance_eval(&host) if host.respond_to?(:call)
```
Comment (joevandyk) [inline lib/webpack/rails/helper.rb]: why not `host = host.call if host.respond_to(:call)` ?

File: lib/webpack/rails/manifest.rb [hunk 3]
```diff
@@ -54,8 +54,8 @@ def load_manifest
 
         def load_dev_server_manifest
           http = Net::HTTP.new(
-            "localhost",
-            ::Rails.configuration.webpack.dev_server.port)
+            ::Rails.configuration.webpack.dev_server.manifest_host,
+            ::Rails.configuration.webpack.dev_server.manifest_port)
           http.use_ssl = ::Rails.configuration.webpack.dev_server.https
           http.verify_mode = OpenSSL::SSL::VERIFY_NONE
           http.get(dev_server_path).body
```
Comment (joevandyk) [inline lib/webpack/rails/manifest.rb]: if there's a problem fetching the manifest, the below error message doesn't look right.. should mention the manifest host/port.

File: README.md (hunk 1 above)
Comment (Aupajo) [inline README.md]: Nothing is yielded to the block, it's just evaluated in the context of the view.

File: lib/webpack/rails/helper.rb (hunk 2 above)
Comment (Aupajo) [inline lib/webpack/rails/helper.rb]: The idea is that the proc is evaluated in the context of the view – so you can use the `request` object and any view helpers you wan.

File: lib/webpack/rails/manifest.rb (hunk 3 above)
Comment (mipearson) [inline lib/webpack/rails/manifest.rb]: You're right - fixed.

**Pull request merged**